### PR TITLE
Add observer macro

### PR DESCRIPTION
**Pull request by bernhardc**
This solves #73.

The macro transforms anonymous wrapped component declarations like
```jsx
const MyComponent = observer(() => { return <div/> })
```
to
```jsx
const MyComponent = observer(function MyComponent() { return <div/> })
```

To use it, the import statement, e.g.
```
import { observer } from "mobx-react-lite";
```
has to be replaced with
```
import { observer } from "mobx-react-lite/dist/macro";
```
The macro touches only the `observer()` function and passes through all others, so users could just always use the `/macro`imports, like
```
import { observer, useComputed } from "mobx-react-lite/dist/macro";
```

Todos:
* Integrate into the build process
Currently I'm using a custom tsconfig.json file (and the code itself is in JS not TS). 
* Make import accessible as `mobx-react-lite/macro`
I din't really know what the best solution woudl be here, maybe have a `macro.js` in the root directory that basically just forwards the export (`module.exports = require('./dist/macro')`)? But then we are still missing the Typescript definitions file.

### CODE REVIEW

File: build-rollup.js [hunk 1]
```diff
@@ -86,13 +88,20 @@ function build(target, mode, filename) {
...
+}
+
 const main = async () => {
     await build("browser", "umd", "index.js")
     await build("browser", "umd.min", "index.min.js")
     await build("browser", "es", "index.module.js")
     await build("native", "cjs", "native.js")
     await build("custom", "umd", "custom.js")
     await build("custom", "es", "custom.module.js")
+    await build("custom", "es", "custom.module.js")
```
Comment (FredyC) [inline build-rollup.js]: I suppose this is duplicated by accident :)

File: package.json [hunk 2]
```diff
@@ -31,15 +31,19 @@
   "homepage": "https://mobx-react.js.org",
   "peerDependencies": {
     "mobx": "^4.0.0 || ^5.0.0",
-    "react": "^16.8.0"
+    "react": "^16.8.0",
+    "babel-plugin-macros": "^2.6.1"
```
Comment (FredyC) [inline package.json]: I think it's better to document this instead of annoying everyone with peer dependency warning.

**Comment by bernhardc** [comment]
OK, I think I found a good solution for the `mobx-react-lite/macro` import: a `/macro/index.js` and `/macro/index.d.ts` that just "forward" to the `/dist` directory.

**Comment by FredyC** [comment]
Sounds good. Would you also write some paragraph to readme about it, please?

**Comment by FredyC** [comment]
So I've replaced Rollup with TSDX, so this will need more tweaking. Probably add some postbuild script to tackle it.

**Comment by FredyC** [comment]
Thanks for the docs update. I am thinking we should probably include some basic tests for this. Would you mind?

**Comment by FredyC** [comment]
So, I am going to be releasing 2.x out soon, most likely by the end of the week. If you can manage to write the test, we can still squeeze it in the 1.x. Given that 2.x is not that big breaking change for most people, the 1.x won't be supported after release. However, it will require a tweaking build process toward TSDX for this PR as I've mentioned before.

**Comment by bernhardc** [comment]
I don't think I have time this week. But no problem, I can migrate the build to TSDX.

**Comment by FredyC** [comment]
@bernhardc Sorry for a delay here. I thought that 2.0 will come out soon, but with revelation of Concurrent mode, I decided to postpone release until it's completely figured out.

Either way, I've just merged TSDX changes to the master. Could you possibly rebase and update code?

**Comment by FredyC** [comment]
If someone ever picks this up again, it will be better to start from scratch.

**Pull request closed without merging**